src/error.rs

```diff
@@ -0,0 +1,186 @@
+            AppErrorKind::RocketError(ref error) => write!(f, "{:?}", error),
+
+            AppErrorKind::InvalidEmailParams(ref error) => {
+                if let Some(error) = error {
+                    write!(f, "{:?}", error)
+                } else {
+                    write!(f, "{}", "Error validating email params.")
+                }
+            }
+
+            AppErrorKind::InvalidProvider(ref provider) => write!(f, "{}", provider),
+            AppErrorKind::ProviderError {
+                ref name,
+                ref description,
+            } => write!(f, "{} error: {:?}", name, description),
+            AppErrorKind::EmailParsingError(ref error) => write!(f, "{:?}", error),
+
+            AppErrorKind::BounceError {
+                ref address,
+                ref bounce,
+                ref description,
+            } => {
+                if let Some(bounce) = bounce {
+                    write!(
+                        f,
+                        "address: {}, bounce: {:?}, description: {}",
+                        address, bounce, description
+                    )
+                } else {
+                    write!(f, "address: {}, description: {}", address, description)
+                }
+            }
+
+            AppErrorKind::DbError(ref error) => write!(f, "{:?}", error),
+
+            AppErrorKind::NotImplemented => write!(f, "{}", "Feature not implemented"),
+        }
+    }
+}
+
+impl AppErrorKind {
+    /// Return a rocket response Status to be rendered for an error
+    pub fn http_status(&self) -> Status {
+        match *self {
+            AppErrorKind::NotFound => Status::NotFound,
+            AppErrorKind::MethodNotAllowed => Status::MethodNotAllowed,
+            AppErrorKind::UnprocessableEntity => Status::UnprocessableEntity,
+            AppErrorKind::TooManyRequests => Status::TooManyRequests,
+            AppErrorKind::InternalServerError => Status::InternalServerError,
+            AppErrorKind::DbError(ref _0) => Status::ServiceUnavailable,
+            AppErrorKind::BadRequest | _ => Status::BadRequest,
+        }
+    }
+}
+
+impl From<AppErrorKind> for AppError {
+    fn from(kind: AppErrorKind) -> AppError {
+        Context::new(kind).into()
+    }
+}
+
+impl From<Context<AppErrorKind>> for AppError {
+    fn from(inner: Context<AppErrorKind>) -> AppError {
+        AppError { inner }
+    }
+}
+
+/// Generate HTTP error responses for AppErrors
+impl<'r> Responder<'r> for AppError {
+    fn respond_to(self, request: &Request) -> response::Result<'r> {
+        let status = self.kind().http_status();
+        let log = MozlogLogger::with_request(request).map_err(|_| Status::InternalServerError)?;
+        slog_error!(log, "{}", &self; "status_code" => status.code,  "status_msg" => status.reason);
+
+        let json = Json(json!({
+            "kind": self.kind().as_ref(),
+            "status": status.code,
+            "message": format!("{}", self)
```

Comment:
We have a standard, pre-existing format for these error payloads, described in the feature doc and also used by the other FxA services (see e.g. `lib/error.js` in `fxa-auth-server` for our definition of `AppError` over there). Let's try to get as close to that as possible.

So `status` should be `code`. Maybe we can map `kind` to `errno`? And we should have separate `error` and `message` fields, where `error` is the standard HTTP error string for each status, and `message` is our more detailed string for the specific error.